src/pages/resultsView/ResultsViewPageHelpers.ts

```diff
@@ -8,15 +8,54 @@ import {CancerStudy} from "../../shared/api/generated/CBioPortalAPI";
 import {VirtualStudy} from "../../shared/model/VirtualStudy";
 import hashString from "../../shared/lib/hashString";
 
+export enum ResultsViewTab {
+    ONCOPRINT="oncoprint",
+    CANCER_TYPES_SUMMARY="cancerTypesSummary",
+    MUTUAL_EXCLUSIVITY="mutualExclusivity",
+    PLOTS="plots",
+    MUTATIONS="mutations",
+    COEXPRESSION="coexpression",
+    ENRICHMENTS="enrichments",
+    SURVIVAL="survival",
+    CN_SEGMENTS="cnSegments",
+    NETWORK="network",
+    EXPRESSION="expression",
+    DOWNLOAD="download"
+}
+
 export function getTabId(pathname:string) {
     const match = pathname.match(/results\/([^\/]+)/);
     if (match) {
-        return match[1];
+        return match[1] as ResultsViewTab;
```

Comment:
seems like it might be safer to just return pathname itself?   also, just fyi, i like the regexp test method b/c it return boolean.   /results\/.test(pathname)

Reply:
pathname is like `/results/plots`, this method extracts 'plots' from that